File: JavaBasic/120_armstrong_num.java
// 120	120_armstrong_num.java	Check whether a number is an Armstrong number (e.g. 153 = 1³+5³+3³).
public class _120_armstrong_num {
    public static void main(String[] args) {
        int num = 153, temp = num, digits = String.valueOf(num).length(), sum = 0;
        while (temp > 0) { sum += (int)Math.pow(temp%10, digits); temp /= 10; }
        System.out.println("Number: " + num);
        System.out.println(sum==num ? num+" is Armstrong." : num+" is NOT Armstrong.");
    }
}

File: JSBASIC/120_armstrong_num.js
// 120	120_armstrong_num.js	Check whether a number is an Armstrong number (e.g. 153 = 1³+5³+3³).

let num = 153;
let temp = num;
let digits = num.toString().length;
let sum = 0;
while (temp > 0) {
    let digit = temp % 10;
    sum += Math.pow(digit, digits);
    temp = Math.floor(temp / 10);
}
console.log("Number:", num);
if (sum === num) {
    console.log(num, "is an Armstrong number.");
} else {
    console.log(num, "is NOT an Armstrong number.");
}
